dotCMS/src/main/java/com/dotmarketing/db/listeners/RollbackListener.java

```diff
@@ -0,0 +1,49 @@
+package com.dotmarketing.db.listeners;
+
+public class RollbackListener implements DotListener {
+    private final String key;
+    private final Runnable listener;
+    public RollbackListener(DotListener listener) {
+        this.key = listener.key();
+        this.listener = listener;
+        
+    }
+    public RollbackListener(String key,Runnable listener) {
+        this.key = key;
+        this.listener = listener;
+        
+    }    
+    
+    @Override
+    public void run() {
+        listener.run();
+        
+    }
+    @Override
+    public String key() {
+        return this.getClass().getName() + ":" + key;
+    }
+
+    @Override
+    public final boolean equals(Object obj) {
+        
+        
+        if(obj == null ||!( obj instanceof RollbackListener)) {
+            return false;
+        }
+        RollbackListener other = (RollbackListener)obj;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'other' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)